microsoft-edge/webview2/concepts/frames.md

```diff
@@ -0,0 +1,500 @@
+* [CoreWebView2Frame.RemoveHostObjectFromScript Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2frame#removehostobjectfromscript) -->
+
+<!-- The above method works like the following method: -->
+<!-- * [CoreWebView2.AddHostObjectToScript Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#addhostobjecttoscript).  See the **Method name** table.  Read both of these API Reference topics, although for frames, you would use the method which supports an `origins` parameter instead. -->
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2Frame::AddHostObjectToScriptWithOrigins method](/microsoft-edge/webview2/reference/win32/icorewebview2frame#addhostobjecttoscriptwithorigins) - has an `origins` parameter.  The documentation doesn't have a **Method name** table.
+
+The above method works like the following method:
+* [ICoreWebView2::AddHostObjectToScript method](/microsoft-edge/webview2/reference/win32/icorewebview2#addhostobjecttoscript).  See the **Method name** table.  Read both of these API Reference topics, although for frames, you would use the method which supports an `origins` parameter instead.
+
+---
+
+<!-- ------------------------------ -->
+#### Example code
+
+See [Step 6: Call AddHostObjectToScript to pass the host object to web-side code](../how-to/hostobject.md#step-6-call-addhostobjecttoscript-to-pass-the-host-object-to-web-side-code) in _Call native-side code from web-side code_.
+
+
+See also:
+* [Host/web object sharing](./overview-features-apis.md#hostweb-object-sharing) in _Overview of WebView2 features and APIs_.
+
+
+<!-- ====================================================================== -->
+## Sending and receiving messages
+
+Messages can be sent between the native app and JavaScript code that's in an iframe:
+*  You can send messages from JavaScript in an iframe in an HTML page to the host app.
+*  You can send messages from the host app to JavaScript in an iframe in an HTML page.
+
+
+#### Sending web messages from an iframe to the host app
+
+To send web messages from an iframe to the host app, use the `window.chrome.webview.postMessage` method:
+
+```javascript
+window.chrome.webview.postMessage(`SetTitleText ${titleText.value}`);
+```
+
+To receive these messages in the host app, the host app must subscribe to the `WebMessageReceived event`.
+
+
+#### Sending messages from the host app to the iframe
+
+The host app sends messages to the iframe by calling the `PostWebMessageAsJson` or `PostWebMessageAsString` method.
+
+The iframe receives the message by subscribing to the `window.chrome.webview.addEventListener('message')` event, as follows:
+
+```javascript
+window.chrome.webview.addEventListener('message', arg => {
+    // implement event listener here
+});
+```
+    
+Code in a webpage might be running either inside a browser or inside the WebView2 control.  Some features might be disabled when inside a WebView2 control instance.
```

Comment:
`Code in a webpage might be running either inside a browser or inside the WebView2 control.  Some features might be disabled when inside a WebView2 control instance.` - from where is it and what does it mean?

Reply:
Removed paragraph.  It is not about iframe - it is super top-level about WebView2 behavior compared to Edge browser.